src/Planners/src/ContactPhaseList.cpp

```diff
@@ -0,0 +1,199 @@
+/**
+ * @file ContactPhase.cpp
+ * @authors Stefano Dafarra
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <BipedalLocomotion/Planners/ContactPhaseList.h>
+
+#include <iostream>
+#include <map>
+#include <cassert>
+#include <algorithm>
+
+using namespace BipedalLocomotion::Planners;
+
+void BipedalLocomotion::Planners::ContactPhaseList::createPhases()
```

Comment:
It reminds me the `unicycle-planner` :+1:

Reply:
This method is a bit obscure (I think I have to go deeper into the `Planners` library before understanding the details).

Could you give me a short recap of what the function is aim to do? Thanks

Reply:
Actually, after your comment on using the ``unordered_map`` instead of the ``vector`` it got simplified. It works as follows. It is used to determine the phases given the contact lists. A phase is characterized by a set of contacts which remains constant (the set) for the entirety of its duration. 

To compute them, I use two auxiliary maps. The items are ordered according to the time (hence the key is the time). I have one for the activations and one for the deactivations. Basically, an element contains all the contacts that got activated or deactivated for a given instant (the key). Then I start iterating. 

The first phase will have the activation time equal to the first key in the activations map. The active contacts correspond to the first element. Then I remove the first element in the activations map. Subsequently, I check if the new first key on the activations map is smaller than the first key in the deactivations map. If this is the case, I have found the end of the previous phase and the begin of the new phase. The new active contacts are the previous ones plus those in the new first element in the activations map. Otherwise, if the first element of the deactivations map was occurring before, I have found again the phase end time and begin time of the new phase. The active contacts instead are the previous ones, minus those contained in the first deactivation element. After this passage, I remove the first deactivation element and I iterate until the two maps are empty.